source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,218 @@
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"PlayerName:- " + player.name);
+                foreach (var task in player.myTasks)
+                {
+                    var firstText = task.name;//.Cast<ImportantTextTask>();
+                    PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskName:- " + task.name);
+                    var modname = "junktonull";
+                    var rolename = "junktonull";
+                    if (!IsNull(pinfo, "Mod"))
+                        modname = modifier.Name;
+
+                    if (!IsNull(pinfo, "Role"))
+                        rolename = role.Name;
+
+                    if (!firstText.Contains(modifier.Name) && !firstText.Contains(role.Name) && !firstText.Contains("_Player"))
+                    {
+                        if (task.IsComplete == true)
+                        {
+                            tasksdone++;
+                        }
+                        taskstotal++;
+                    }
+                }
+
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskDone:- " + tasksdone);
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskTotal:- " + taskstotal);
+                pinfo.TasksCompleted = tasksdone;
+                pinfo.TasksTotal = taskstotal;
+                pinfolist.Add(pinfo);
+                tasksdone = 0;
+                taskstotal = 0;
+                AdditionalTempData.playerData = pinfolist;
+            }
+        }
+
+        public static bool HasTasks(Faction faction)
+        {
+            if (faction == Faction.Crewmates)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        static bool IsNull(AdditionalTempData.PlayerInfo pinfo, string type)
+        {
+            if (type == "Role")
+            {
+                if (pinfo.CurrentRole != null)
+                {
+                    return false;
+                }
+                else
+                {
+                    return true;
+                }
+            }
+            else if (type == "Mod")
+            {
+                if (pinfo.Modifier != null)
+                {
+                    return false;
+                }
+                else
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                return true;
+            }
```

Comment:
Why not use switch case?
```suggestion
            return type switch {
                "Role" => pinfo.CurrentRole == null,
                "Mod" => pinfo.Modifier == null,
                _ => true
            };
```